### DISCUSSION TITLE
printing all labels at once

### DISCUSSION THREAD

**Discussion by holle75** [lairdubois/lairdubois-opencutlist-sketchup-extension#665] [Ideas]
Hi, i´m an admirer of opencutlist. I love it. Thanks for the coolest plugin ever ...

This out of the way ... i just spent 3 hours trying to print 330 labels for 15 different types of dimensional wood (15 groups). I gave up.
It´s great to be able to select the offset (to not throw away 22 labels on the sheet you only printed 2 labels from the group before). But when your printer is configured to start with the last page of the next group with 5 pages you have to slip the former printed page at the right position in your paper tray. This is kind of awkward ;) ... not to say opening 15 groups and configure the labels is quite time consuming (even when using a preset).

Ok, my idea: one print-it-all-option (all shown/not hidden groups in the parts list ... or groups you can select when choosing this print-it-all-option)

**Comment by bbeaulant** [comment]
Hello @holle75,

Ok, you win. In 7.0, I will add the ability to generate the labels from the whole parts list. 

<img width="1262" alt="Capture d’écran 2024-10-16 à 10 03 45" src="https://github.com/user-attachments/assets/3c23c699-e687-416f-a995-cd3cb00b1a3a">

The feature will work like "Draw" with its private settings. So : 
- this only allows to have a single layout for all groups
- this doesn't allow to sort labels by bin in any way
- Labels will be grouped by "group" in the same order than in parts list

---

Else, I'm quite curious the test a project with as much groups. Can you share it with me ?

**Reply by holle75** [reply]
Dear Boris, that looks promising and fits my intentions! Great!

I’m away for 10 days. Let me see after how I can „destroy“ this „confidential“ project that it still be useful in OCL/for you, but wouldn’t put me out of my job ;)

**Comment by holle75** [comment]
Hi Boris, i´m back. I was trying to "flatten" the project without success (only way i found was exporting all geometry and import it again. This works nicely, but all material is gone as suspected).

Do you have any idea how i can provide you with what you need without giving the opportunity to "copy" the project? This is not supposed to sound harsh, but i´m not allowed to send this project "as is".

Anyhow, wouldn´t it be more easy if you just create some dummy materials/components to reach the same goal?

**Reply by bbeaulant** [reply]
If you can't share it, too bad. It was interesting with a **real** project to analyze how OCL can give the best answer.
With a dummy project, it makes less sense.

**Comment by holle75** [comment]
I can share it if you can suggest a way to "flatten" all components on the ground plane for example. Or whatever other idea to make the project "unusable" besides for OCL.
